pandas/io/excel/_base.py

```diff
@@ -691,7 +691,8 @@ class ExcelWriter(metaclass=abc.ABCMeta):
 
         .. versionadded:: 1.3.0
     engine_kwargs : dict, optional
-        Keyword arguments to be passed into the engine.
+        Keyword arguments to be passed into the engine. Only works for
+        xlsxwriter, openpyxl and xlwt. Does not work in append mode.
```

Comment:
the only other writer is odswriter which doesn't have any kwargs and I got errors when adding them.  It is confusing in append mode, because there the book is loaded with `load_workbook`, whereas in normal writer mode, it is `Workbook. However, I do think it should also be passed through in `a` mode, because there are useful arguments there such as `"keep_vba":True`

Reply:
I now added this mention in an example

Reply:
> the only other writer is odswriter which doesn't have any kwargs and I got errors when adding them

Can you explain this? Are you getting errors even when kwargs is empty? It seems preferable to pass through kwargs regardless - if odswriter gains arguments then we don't need to do anything. Also, if a user tries to pass in invalid kwargs to the engine, the code should fail.

> It is confusing in append mode, because there the book is loaded with load_workbook, whereas in normal writer mode, it is Workbook.

Agreed it is confusing, but I think we should instead document that different methods are called when creating a new workbook vs appending to one. Adding a note in the argument documentation along the lines of

> When using append mode on an existing file, pandas will pass ``engine_kwargs`` to the engine function that loads the workbook.
> - openpyxl: openpyxl.reader.excel.load_workbook
> etc.

Does that make sense?

Reply:
The errors were due to `combine_kwargs` making a dictionary form `None` in all other engines, implemented now.

I now did it all the way explicit like so:
![image](https://user-images.githubusercontent.com/33122845/133685964-ca306dd0-5e56-4784-b1bb-581ba1760be6.png)

Reply:
Looks great - thanks!